java-squid/src/main/java/org/sonar/java/resolve/BytecodeCompleter.java

```diff
@@ -144,15 +144,24 @@ public String formFullName(String name, JavaSymbol site) {
 
   // FIXME(Godin): or parameter must be renamed, or should not receive flat name, in a former case - first transformation in this method seems useless
   JavaSymbol.TypeJavaSymbol getClassSymbol(String bytecodeName, int flags) {
+    return getClassSymbol(null, bytecodeName, flags);
+  }
+  public JavaSymbol.TypeJavaSymbol getClassSymbol(@Nullable JavaSymbol.TypeJavaSymbol classSymbolOwner, String bytecodeName, int flags) {
     String flatName = Convert.flatName(bytecodeName);
     JavaSymbol.TypeJavaSymbol symbol = classes.get(flatName);
     if (symbol == null) {
       String shortName = Convert.shortName(flatName);
       String packageName = Convert.packagePart(flatName);
-      String enclosingClassName = Convert.enclosingClassName(shortName);
-      if (StringUtils.isNotEmpty(enclosingClassName)) {
+      JavaSymbol.TypeJavaSymbol owner = classSymbolOwner;
+      if(owner == null) {
+        String enclosingClassName = Convert.enclosingClassName(shortName);
+        if(StringUtils.isNotEmpty(enclosingClassName)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "classSymbolOwner". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)